core/src/apps/common/sdcard.py

```diff
@@ -45,17 +45,57 @@ class SdProtectCancelled(wire.ProcessError):
     return await confirm(ctx, text, confirm=btn_confirm, cancel=btn_cancel)
 
 
+async def format_card_dialog(ctx: wire.GenericContext) -> bool:
+    # Format card? yes/no
+    text = Text("SD card error", ui.ICON_WRONG, ui.RED)
+    text.bold("Unknown filesystem.")
+    text.br_half()
+    text.normal("Use a different card or")
+    text.normal("format the SD card to")
+    text.normal("the FAT32 filesystem.")
+    if not await confirm(ctx, text, confirm="Format", cancel="Cancel"):
+        return False
+
+    # Confirm formatting
+    text = Text("Format SD card", ui.ICON_WIPE, ui.RED)
+    text.normal("Do you really want to", "format the SD card?")
+    text.br_half()
+    text.bold("All data on the SD card", "will be lost.")
+    return await hold_to_confirm(ctx, text, confirm="Format SD card")
+
+
 async def sd_problem_dialog(ctx: wire.GenericContext) -> bool:
-    text = Text("SD card protection", ui.ICON_WRONG, ui.RED)
+    text = Text("SD card problem", ui.ICON_WRONG, ui.RED)
     text.normal("There was a problem", "accessing the SD card.")
     return await confirm(ctx, text, confirm="Retry", cancel="Abort")
 
 
-async def ensure_sd_card(ctx: wire.GenericContext) -> None:
-    sd = io.SDCard()
-    while not sd.present():
+async def ensure_sdcard(ctx: wire.GenericContext) -> None:
```

Comment:
We should use a universal symbol IMHO, I went with `sdcard` because it's shorter. No strong preference though, we can switch to `sd_card` in module names, variables, etc.

Reply:
note that symbols in C are also always `sdcard` with no underscore

Reply:
Ok, let's not keen on it. I find it a bit strange we use underscore somewhere and somewhere not, but have it in line with C makes sort of sense.